src/main/java/edu/ucdavis/dss/ipa/services/jpa/JpaTeachingAssignmentService.java

```diff
@@ -140,19 +140,9 @@ public TeachingAssignment findByTeachingAssignment(TeachingAssignment teachingAs
 
 	@Override
 	public List<TeachingAssignment> findAllByIds(List<Long> teachingAssignmentIds) {
-		List<TeachingAssignment> teachingAssignments = new ArrayList<>();
+		List<TeachingAssignment> teachingAssignments = teachingAssignmentRepository.findByIdIn(teachingAssignmentIds);
 
-		boolean allIdsValid = true;
-
-		for(Long id : teachingAssignmentIds) {
-			TeachingAssignment teachingAssignment = this.findOneById(id);
-			if (teachingAssignment == null) {
-				allIdsValid = false;
-				break;
-			}
-		}
-
-		if (allIdsValid == false) {
+		if (teachingAssignments.size() != teachingAssignmentIds.size()) {
```

Comment:
Should this function be doing this check? It seems like findAllByIds() simply finds all by IDs. I'd favor failing gracefully here, i.e. if you can only find 5 of 6, return 5, and putting the size check logic in the caller or whoever cares about that.

Reply:
I was kinda torn on that separation, but I took the perspective that if some code handed this method Ids that don't exist in the database, then there is definitely a bug. It seemed good to ensure that sanity checking always occurs. 

The more I think about it, this is kind of a 'verifying the param data is sane', and that does fall under the responsibility of the controller.

Ok you've convinced me, will move it.

Reply:
I see. My only additional thought would be if you _didn't_ want to move the logic, the name should somehow indicate that's how it works because otherwise, I think of it like SQL: find all these rows with the following IDs ... if an ID is missing, SQL doesn't error out, it just does its best.